jina/types/sets/cache.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+from .document import DocumentSet
+
+
+class CacheDocumentSet:
+
+    def __init__(self,
+                 capacity: Optional[int] = None,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.capacity = capacity
+        self._doc_set = DocumentSet()
```

Comment:
We could also have an `id` for the `CacheDocumentSet` so that when a new set is created because `left capacity` is smaller than number of documents, the new `CacheSet` for the overflow has a subsequent identifier than the previous `CacheSet`.

Reply:
I will have it private for `EncodeDriver` because it is very specific to help in this functionality, do not want to provide identifiers or have it be a primary type for Jina